src/StockAvailabilityChecker.php

```diff
@@ -62,7 +63,13 @@ public function check(PurchasableEntityInterface $entity, $quantity, Context $co
       $stock_config->getLocationList($entity)
     );
 
-    return ($stock_level >= $quantity);
+    if ($stock_level >= $quantity) {
+      return AvailabilityResponse::available(0, $stock_level);
+    }
+    else {
+      return AvailabilityResponse::unavailable(0, $stock_level, 'has Insufficient stock');
```

Comment:
we should let the stock service determine the text for "has Insufficient stock" but that can go after.